src/containers/edit-profile/professional-info-tab/add-professional-category-modal/AddProfessionalCategoryModal.tsx

```diff
@@ -112,36 +116,38 @@ const AddProfessionalCategoryModal: FC<AddProfessionalCategoryModalProps> = ({
 
   const formSubmission = () => {
     const userRoleCategory = userRole as MainUserRole
-    const { category } = data
-
-    // TODO: icon should be displayed accordingly to category
+    const { category, subjects } = data
     if (category.appearance === undefined) {
       category.appearance = { color: '#E3B21C', icon: 'ScienceRoundedIcon' }
     }
-
+    const sanitizedCategory = {
+      ...category,
+      name: t(`categories.${category.name}`, {
+        lng: 'en',
+        defaultValue: category.name
+      })
+    }
+    const sanitizedSubjects = subjects.map((subject) => ({
+      ...subject,
+      name: t(`subjects.${subject.name}`, {
+        lng: 'en',
+        defaultValue: subject.name
+      })
+    }))
+    const categoryData: UserMainSubject = {
+      ...data,
+      category: sanitizedCategory,
+      subjects: sanitizedSubjects,
+      _id: isEdit ? (initialValuesFromProps?._id ?? '') : uuidv4(),
```

Comment:
Consider using native browser `crypto.randomUUID()` method instead of `uuidv4()`.